audbackend/core/backend.py

```diff
@@ -455,26 +417,35 @@ def put_file(
 
         Raises:
             FileNotFoundError: if local file does not exist
+            ValueError: if ``dst_path`` contains invalid character
+            ValueError: if ``dst_path`` does not end on file extension
 
         """
+        dst_path, ext = utils.check_path_and_ext(dst_path, ext)
         if not os.path.exists(src_path):
-            raise FileNotFoundError(
-                errno.ENOENT, os.strerror(errno.ENOENT), src_path,
-            )
+            utils.raise_file_not_found_error(src_path)
 
-        dst_path = self.path(dst_path, version, ext=ext)
+        skip = False
 
-        # skip if file with same checksum exists on backend
-        skip = self._exists(dst_path) and \
-            utils.md5(src_path) == self._checksum(dst_path)
-        if not skip:
-            self._put_file(src_path, dst_path, verbose)
+        # skip if file with same checksum already exists
+        if self._exists(dst_path, version, ext):
+            checksum = self._checksum(dst_path, version, ext)
+            skip = utils.md5(src_path) == checksum
 
-        return dst_path
+        if not skip:
```

Comment:
Maybe the following is easier to understand?

```python
# skip if file with same checksum already exists
if not (
    self._exists(dst_path, version, ext)
    and utils.md5(src_path) == self._checksum(dst_path, version, ext)
):
```

Then you could also remove
```python
skip = False
```

Reply:
Yes, makes sense. Changed it accordingly.